extension_templates/forecasting.py

```diff
@@ -384,7 +384,7 @@ def _predict_var(self, fh, X=None, cov=False):
     #   implementation which uses variance or quantiles under normal assumption
     #
     # if not implementing, delete _predict_proba
-    def _predict_proba(self, fh, X):
+    def _predict_proba(self, fh, X, marginal=True):
         """Compute/return fully probabilistic forecasts.
 
         private _predict_proba containing the core logic, called from predict_proba
```

Comment:
Not in the current PR, but could you kindly update docstr, the lines below are unclear to me and aren't according to docstr standard. `X: one of self.get_tag("X_inner_mtype"), optional, default=None`, `marginal: bool, optional, default=True`. What does "guaranteed to be ForecastingHorizon" mean?

Reply:
hm, can you be precise what should be improved with the docstring? I agree that it might be obscure and should be improved, but is there an additional problem?
Will fix in separate PR.

> "guaranteed to be ForecastingHorizon"

The underscore function is a function that is called only by other code, never by the user.
"guaranteed" in such a context means: those other functions only call it with certain inputs (instance of `ForecastingHorizon`), and that is guaranteed by the architecture, logic, tests, etc.

Reply:
I've updated the docstrings in this PR which also updates the error message:
https://github.com/alan-turing-institute/sktime/pull/2314

I hope this is clearer?

I've separated PR to keep the different changes compartmentalized.